include/engine/api/route_api.hpp

```diff
@@ -228,42 +228,51 @@ class RouteAPI : public BaseAPI
 
         std::vector<util::json::Object> annotations;
 
-        if (parameters.annotations_type != RouteParameters::AnnotationsType::None)
+        if (parameters.annotations_type != RouteParameters::AnnotationsType::None ||
+            parameters.annotations == true)
         {
+            auto requested_annotations = parameters.annotations_type;
+            if ((parameters.annotations == true) &&
+                (parameters.annotations_type == RouteParameters::AnnotationsType::None))
+            {
+                requested_annotations = RouteParameters::AnnotationsType::All;
+            }
```

Comment:
If this code block is moved before line 231 you can change the first if to also only check for the `::None` case. That reduces the points in the code that need to know about the work-a-round.